lib/experimental/blocks.php

```diff
@@ -121,3 +121,69 @@ function gutenberg_register_metadata_attribute( $args ) {
 	return $args;
 }
 add_filter( 'register_block_type_args', 'gutenberg_register_metadata_attribute' );
+
+
+$gutenberg_experiments = get_option( 'gutenberg-experiments' );
+if ( $gutenberg_experiments && array_key_exists( 'gutenberg-connections', $gutenberg_experiments ) ) {
+	/**
+	 * Renders the block meta attributes.
+	 *
+	 * @param string   $block_content Block Content.
+	 * @param array    $block Block attributes.
+	 * @param WP_Block $block_instance The block instance.
+	 */
+	function gutenberg_render_custom_sources( $block_content, $block, $block_instance ) {
+		$meta_custom_source = require __DIR__ . '/custom-sources/meta.php';
```

Comment:
I'm a bit confused about the "custom sources" naming. Is this what we are doing? Are we rendering custom sources? If I understood it correctly, we are:

* Processing the connections block attribute.
* Getting the value from the connections source. Meta in this case using `get_post_meta`.
* Replacing the HTML according to the block attributes source. `html` in the case of the paragraph.

In my opinion, this last step should be independent of the connections source. If I get the value from a different source, I would still have to replace the same HTML, but with a different value. Is this right?

Just to have more context, I believe in the future this distinction will be more relevant:

* **Connections sources**: From where we get the value. Right now it is only meta, but it could be drupal, parent-block, my-custom-database...
* **Block attributes source**: Dictates which part of the HTML needs to be replaced. Thanks to the block attributes source, it's possible to know the HTML we have to replace. If it is `html` we will replace the inner content. If it is `attribute` we will replace an HTML attribute.

That's why it makes sense to separate the code as well.

Reply:
> * Processing the connections block attribute.
> * Getting the value from the connections source. Meta in this case using `get_post_meta`.
> * Replacing the HTML according to the block attributes source. `html` in the case of the paragraph.

You're right, that's the right abstraction of the 3 steps that we perform. As it stands right now, the 2nd & 3rd step are not separated. It makes sense that they should be.

Also, some of the naming is a direct copy from Riad's PR (https://github.com/WordPress/gutenberg/pull/51375) which is not 100% accurate anymore. 

I ll change it.

Reply:
The code should be better abstracted now to reflect those 3 separate steps. 1 & 3 are both inside of the same function but we can abstract it further when we add support for the Image block & other attributes. Then we can abstract the part that replaces the different attribute sources (`html`, `attribute`, etc.).

We should discuss consistent naming though of Step 2. Should it be:
- custom sources
- custom connections
- connection sources
- custom connection sources
- ...something else?

Reply:
Yes, I believe the whole naming should be discussed. Another option I was thinking of was using "Dynamic data" instead of "Connections". So it could be "Dynamic data sources". But I didn't think enough about it. What doesn't quite fit me is the use of "custom". What does "custom" mean in this context? Which sources are not custom?

Do we need it for this initial experiment? Is it too difficult to change it afterwards?

Reply:
> Is it too difficult to change it afterwards?
Nope, it won't be difficult at all, I just wanted to start this conversation and have consistent naming even if it gets changed later 🙂 .

I guess "custom" means something like "different than whatever WP uses to get the content normally" but I agree that it's a bit meaningless.

I like "Connections" a little bit better than "Dynamic Data" because the latter can mean many things in many different contexts. Most data is dynamic rather than static in programming 🤓. I think "Connection Sources" sounds best so far. "Connections" alone is also a bit vague. "Connection Sources" implies "_data_ sources" which is roughly what we are referring to here.

Reply:
Maybe "block connections" is also a good name? We're **connecting stuff to blocks** after all.

Reply:
Yep, I was thinking about names using this sentence: **we are connecting block attributes to values that might change depending on the context (post_id, parent block…)**. But not sure yet which is the best abstraction 😄

Reply:
Hmm, I would just go with "block connections" as a name for the whole thing (like I use it in https://github.com/WordPress/gutenberg/pull/53247/files#diff-be8a354ce9c01e85c0fdf04c75d76880cdc6c81992db7a86a7a0515dc11a9833R135) and "connection sources" for the thing that does the connecting to meta fields/parent blocks/other DBs, etc.

We can always change it in the future.

If you don't object, then I ll go ahead and change it.

Reply:
Sure! We can always change the naming later if we agree on a different one.